src/mbgl/tile/geometry_tile_data.hpp

```diff
@@ -73,4 +74,72 @@ Feature convertFeature(const GeometryTileFeature&, const CanonicalTileID&);
 // The result is guaranteed to have correctly wound, strictly simple rings.
 GeometryCollection fixupPolygons(const GeometryCollection&);
 
+struct ToGeometryCollection {
+    GeometryCollection operator()(const mapbox::geometry::point<int16_t>& geom) const {
+        GeometryCollection collection;
+        GeometryCoordinates coordinates;
+        coordinates.emplace_back(geom.x, geom.y);
+        collection.push_back(coordinates);
+        return collection;
+    }
+    GeometryCollection operator()(const mapbox::geometry::multi_point<int16_t>& geom) const {
+        GeometryCollection collection;
+        GeometryCoordinates coordinates;
```

Comment:
As a performance optimization, wherever you copy vector-to-vector with `emplace_back` in a loop, do `coordinates.reserve(geom.size())` beforehand to pre-allocate the exact size.